frontend/components/BulkActionSelector/BulkActionSelector.jsx

```diff
@@ -72,32 +73,49 @@ export default class BulkActionSelector extends Component {
     const containerStyle = new Style(styles, 'container')
       .addIf('container-empty', selection.length === 0)
       .addResolved(className)
-    let placeholder = 'With selected'
+    const placeholder = 'Bulk actions'
 
-    if (selection.length > 0) {
-      placeholder += ` (${selection.length})`
+    // Generate new actions object for the dropdown options
+    let modifiedActions = {}
+    Object.keys(actions).forEach(key => {
+      modifiedActions[key] = actions[key].label
+    })
+
+    let button
+
+    // If the option comes with a confirmation message, display
+    // ButtonWithPrompt, otherwise a standard Button is used.
+    if (selected !== ACTION_PLACEHOLDER && actions[selected].ctaMessage) {
```

Comment:
This is just a matter of personal preference, but I would create a Boolean outside of the JSX block that indicates what type of button we need and then use that as a conditional. That way the markup stays all together and the whole thing reads easier to me.

Like this:

```jsx
  render() {
    const {
      actions,
      className,
      onChange,
      selection = []
    } = this.props
    const {selected} = this.state
    const containerStyle = new Style(styles, 'container')
      .addIf('container-empty', selection.length === 0)
      .addResolved(className)
    const placeholder = 'Bulk actions'

    // Generate new actions object for the dropdown options
    let modifiedActions = {}
    Object.keys(actions).forEach(key => {
      modifiedActions[key] = actions[key].label
    })

    // If the option comes with a confirmation message, display
    // ButtonWithPrompt, otherwise a standard Button is used.
    const needsButtonWithPrompt = selected !== ACTION_PLACEHOLDER &&
      actions[selected].ctaMessage

    return (
      <div class={containerStyle.getClasses()}>
        <DropdownNative
          className={styles.dropdown}
          onChange={this.onChange.bind(this)}
          options={modifiedActions}
          placeholderLabel={placeholder}
          placeholderValue={ACTION_PLACEHOLDER}
          textSize="small"
          value={selected}
        />

        {needsButtonWithPrompt && (
          <ButtonWithPrompt
            accent="data"
            disabled={(selected === ACTION_PLACEHOLDER) || actions[selected].disabled}
            onClick={this.onApply.bind(this)}
            promptCallToAction={actions[selected].ctaMessage}
            promptMessage={actions[selected].confirmationMessage}
            size="small"
          >Apply</ButtonWithPrompt>          
        )}

        {!needsButtonWithPrompt && (
          <Button
            accent="data"
            disabled={selected === ACTION_PLACEHOLDER}
            onClick={this.onApply.bind(this)}
            size="small"
          >Apply</Button>
        )}
      </div>
    )
  }
```